### ISSUE TITLE
str_replace() silently fails with array arguments for `from` and `to`

### ISSUE THREAD

**Issue by MarkBaker** [zephir-lang/zephir#1089]
The following code is a literal translation of PHP code using arrays for the `from` and `to` arguments in a `str_replace()` call, which is perfectly valid syntax in PHP:

```
let complexNumber =  "1.23e-4--2.34e-5i";
var_dump(complexNumber);

// Fix silly human errors
let complexNumber = str_replace(["+-", "-+", "++", "--"], ["-", "-", "+", "+"], complexNumber);
var_dump(complexNumber);
```

gives an output of

```
string(17) "1.23e-4--2.34e-5i"
NULL
```

It's only because I'm var_dumping the results that I see that the `str_replace()` call has completely failed, without even a warning

---

While I can simulate the from/to array behaviour using

```
private fromReplace = ["+-", "-+", "++", "--"];
private toReplace = ["-", "-", "+", "+"];
...
let complexNumber =  "1.23e-4--2.34e-5i";
var_dump(complexNumber);

// Fix silly human errors
for key, value in this->fromReplace {
    let complexNumber = str_replace(value, this->toReplace[key], complexNumber);
}
var_dump(complexNumber);
```

this feels awkwardly unnecessary, and still doesn't explain the lack of warning when using arrays for the from/to arguments

**Comment by sergeyklay** [comment]
@MarkBaker

Fixed in current master.

``` zephir
namespace Utils;

class Greeting
{
    private fromReplace = ["+-", "-+", "++", "--"];
    private toReplace = ["-", "-", "+", "+"];

    public static function fixComplex1()
    {
        var complexNumber;
        let complexNumber = "1.23e-4--2.34e-5i";

        return str_replace(["+-", "-+", "++", "--"], ["-", "-", "+", "+"], complexNumber);
    }

    public static function fixComplex2()
    {
        var complexNumber;
        array search = ["+-", "-+", "++", "--"];
        array replace = ["-", "-", "+", "+"];

        let complexNumber = "1.23e-4--2.34e-5i";

        return str_replace(search, replace, complexNumber);
    }

    public function fixComplex3()
    {
        var complexNumber, fromReplace, toReplace, key, value;

        let complexNumber =  "1.23e-4--2.34e-5i";
        let fromReplace = this->fromReplace;
        let toReplace = this->toReplace;

        for key, value in fromReplace {
            let complexNumber = str_replace(value, toReplace[key], complexNumber);
        }

        return complexNumber;
    }
}
```

``` sh
$ php -r 'echo Utils\Greeting::fixComplex1(), PHP_EOL;'
1.23e-4+2.34e-5i

$ php -r 'echo Utils\Greeting::fixComplex2(), PHP_EOL;'
1.23e-4+2.34e-5i

$ php -r '$u = new Utils\Greeting; echo $u->fixComplex3(), PHP_EOL;'
1.23e-4+2.34e-5i
```

**Comment by MarkBaker** [comment]
Thanks for that.
I'm still using the tagged 0.7.1 release, but I've put a TODO in my code noting that it will be available when I move on to 0.8
